app/consumer/genesis_test.go

```diff
@@ -170,52 +424,247 @@ func getGenesisTransformCmd() (*cobra.Command, error) {
+	require.EqualValues(t, srcGenesis.Params.DistributionTransmissionChannel, resultGenesis.Params.DistributionTransmissionChannel)
+	require.EqualValues(t, srcGenesis.Params.ProviderFeePoolAddrStr, resultGenesis.Params.ProviderFeePoolAddrStr)
+	require.EqualValues(t, srcGenesis.Params.CcvTimeoutPeriod, resultGenesis.Params.CcvTimeoutPeriod)
+	require.EqualValues(t, srcGenesis.Params.TransferTimeoutPeriod, resultGenesis.Params.TransferTimeoutPeriod)
+	require.EqualValues(t, srcGenesis.Params.ConsumerRedistributionFraction, resultGenesis.Params.ConsumerRedistributionFraction)
+	require.EqualValues(t, srcGenesis.Params.HistoricalEntries, resultGenesis.Params.HistoricalEntries)
+	require.EqualValues(t, srcGenesis.Params.UnbondingPeriod, resultGenesis.Params.UnbondingPeriod)
+	require.EqualValues(t, srcGenesis.Params.SoftOptOutThreshold, resultGenesis.Params.SoftOptOutThreshold)
+	require.EqualValues(t, srcGenesis.Params.RewardDenoms, resultGenesis.Params.RewardDenoms)
+	require.EqualValues(t, srcGenesis.Params.ProviderRewardDenoms, resultGenesis.Params.ProviderRewardDenoms)
+
+	require.Equal(t, srcGenesis.ProviderClientState, resultGenesis.Provider.ClientState)
+	require.Nil(t, resultGenesis.ProviderClientState)
+
+	require.Equal(t, srcGenesis.Provider.ConsensusState, resultGenesis.ProviderConsensusState)
+	require.Nil(t, resultGenesis.ProviderConsensusState)
+
+	require.Equal(t, srcGenesis.NewChain, resultGenesis.NewChain)
+	require.Equal(t, "", resultGenesis.ProviderClientId)
+	require.Equal(t, "", resultGenesis.ProviderChannelId)
+	require.Equal(t, srcGenesis.InitialValSet, resultGenesis.Provider.InitialValSet)
+	require.Empty(t, resultGenesis.InitialValSet)
+
+}
+
+// Check transformation of provider v3.3.x implementation to consumer V2
+func TestConsumerGenesisTransformationV330ToV2(t *testing.T) {
+	version := "v3.3.x"
+	filePath := createConsumerDataGenesisFile(t, version)
 	defer os.Remove(filePath)
 
-	cmd, err := getGenesisTransformCmd()
-	require.NoError(t, err, "Error setting up transformation command: %s", err)
-	cmd.SetArgs([]string{filePath})
+	var srcGenesis consumerTypes.GenesisState
+	ctx := getClientCtx()
+	err := ctx.Codec.UnmarshalJSON([]byte(consumerGenesisStates[version]), &srcGenesis)
+	require.NoError(t, err)
 
-	output := new(bytes.Buffer)
-	cmd.SetOutput(output)
+	targetVersion := "v2.x"
+	result, err := transformConsumerGenesis(filePath, &targetVersion)
 	require.NoError(t, err)
-	_, err = cmd.ExecuteC()
+
+	resultGenesis := consumerTypes.GenesisState{}
+	err = ctx.Codec.UnmarshalJSON(result, &resultGenesis)
 	require.NoError(t, err)
 
-	var oldConsumerGenesis map[string]interface{}
-	err = json.Unmarshal([]byte(consumerGenesisStates[version]), &oldConsumerGenesis)
-	require.NoError(t, err, "Error parsing old version of ccv genesis content for consumer")
+	require.Equal(t, srcGenesis.Params, resultGenesis.Params)
```

Comment:
Similar here - I think here it's probably harder and might not be worth it.
Technically, I think we could provide old binaries inside the repo and use those to parse the genesis to check that it actually works with the old version, but probably it's too much effort and what you are doing is fine

Reply:
same as above... tried to do so but failed